src/hardware/gus.cpp

```diff
@@ -209,6 +210,7 @@ using voice_array_t = std::array<std::unique_ptr<Voice>, MAX_VOICES>;
 class Gus {
 public:
 	Gus(uint16_t port, uint8_t dma, uint8_t irq, const std::string &dir);
+	~Gus();
```

Comment:
Destructor must be `virtual`.

Reply:
Just for my own understanding, is this because the destructor now gets wrapped in an `unique_ptr` type?
(otherwise, Gus is it's own object, and nothing derives from it)

Reply:
> Just for my own understanding, is this because the destructor now gets wrapped in an unique_ptr type?

No.

> otherwise, Gus is its own object, and nothing derives from it

Yes, but if potentially someone *would* derive from Gus class, then it's very likely that implementer would forget to double-check if base class has virtual destructor - deriving class should not require a modification of base class (Open-Closed principle in OO programming).

We could invent some complicated style guide where to use virtual destructor and when to avoid it, but it's better to have a simple rule of thumb:

- every non-default constructor should be virtual - either by marking it `virtual` (if it's a base class) or by marking it `override` (which implies virtual destructor in base class and forces compiler check).

Reply:
Thanks for the clear explanation, and makes sense. Deriving a class should certainly not require changing the base class!

Regarding the rule of thumb - _"every non-default constructor should be ... "_ (did you mean destructor)?

Reply:
Ah, yes - I meant d-tor :)